CA_DataUploaderLib/Extensions/StringExtensions.cs

```diff
@@ -31,7 +31,8 @@ public static string StringBetween(this string s, string match1, string match2)
         public static double ToDouble(this string s) => double.Parse(s, NumberStyles.Float, CultureInfo.InvariantCulture);
         public static bool TryToDouble(this string s, out double val) => double.TryParse(s, NumberStyles.Float, CultureInfo.InvariantCulture, out val);
         public static bool TryToDouble(this ReadOnlySpan<char> s, out double val) => double.TryParse(s, NumberStyles.Float, CultureInfo.InvariantCulture, out val);
-        public static List<string> SplitNewLine(this string s) => s.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries).ToList();
+        public static List<string> SplitNewLine(this string s, bool removeEmptyEntries = true) => s.Split(new[] { Environment.NewLine }, removeEmptyEntries ? StringSplitOptions.RemoveEmptyEntries : StringSplitOptions.None).ToList();
```

Comment:
I know we had discussed this one, but  an alternative is to take the optional StringSplitOptions instead of the boolean and default it to RemoveEntries

Reply:
done